session/pessimistic_test.go

```diff
@@ -1727,3 +1727,28 @@ func (s *testPessimisticSuite) TestAmendTxnVariable(c *C) {
 	// Restore.
 	tk2.MustExec("set global tidb_enable_amend_pessimistic_txn = 1;")
 }
+
+func (s *testPessimisticSuite) TestSelectForUpdateWaitSeconds(c *C) {
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk.MustExec("drop table if exists tk")
+	tk.MustExec("create table tk (c1 int primary key, c2 int)")
+	tk.MustExec("insert into tk values(1,1),(2,2),(3,3),(4,4),(5,5)")
+	tk2 := testkit.NewTestKitWithInit(c, s.store)
+	tk3 := testkit.NewTestKitWithInit(c, s.store)
+
+	// tk2 lock c1 = 1
+	tk2.MustExec("begin pessimistic")
+	tk3.ExecToErr("begin pessimistic")
+	tk2.MustExec("select * from tk where c1 = 1 for update")
+	errCh := make(chan error, 1)
+	go func() {
+		// tk3 try lock c1 = 1 timeout 1sec, the default innodb_lock_wait_timeout value is 50s.
+		err := tk3.ExecToErr("select * from tk where c1 = 1 for update wait 1")
+		tk3.ExecToErr("rollback")
+		errCh <- err
+	}()
+	time.Sleep(time.Millisecond * 1500)
```

Comment:
In the CI environment, the timing is not accurate. This may introduce an unstable test.